util/image-canonical-ref/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/types"
+)
+
+// This is a simple tool to resolve canonical reference of the image.
+// E.g. this resolves quay.io/operator-framework/olm:v0.28.0 to
+// quay.io/operator-framework/olm@sha256:40d0363f4aa684319cd721c2fcf3321785380fdc74de8ef821317cd25a10782a
+func main() {
+	ctx := context.Background()
+
+	if len(os.Args) != 2 {
+		fmt.Fprint(os.Stderr, "missing argument: provide image reference\n")
```

Comment:
nit: if we have > 2 arguments, the message could be misleading. Maybe something like:
```suggestion
		fmt.Fprint(os.Stderr, fmt.Sprintf("usage: %s <image reference>\n", args[0])")
```
?